### PR TITLE
Testing using Rspec

### PR DESCRIPTION
**Pull request by TimmyChan99**
In this PR we:
- Write unit tests using Rspec.
-  Add unit tests to classes:
    - `Book`
     - `Person`
    - `Student`
    - `Teacher`
    - `Classroom`
    - `Decorator`
    - `Rental`

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    expect(@person.age).to eq 10
+  end
+
+  it 'Should return person name' do
+    expect(@person.name).to eq 'Student1'
+  end
+
+  # it "Should return person's parent_permission" do
+  # 	expect(@person.parent_permission) .to be true
+  # end
```
Comment (Deepakdanger) [inline spec/person_spec.rb]: - [x] I would recommend removing the commented code to make the codebase cleaner.

- [x] I would recommend adding test for all the public method you have for the respective class. 

Hint: 
- Add test for `can_use_services?`, `correct_name`, and `add_rental` method.

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  before :each do
+    @book = Book.new 'Title', 'Author'
+  end
+
+  it 'It should return book title' do
+    expect(@book.title).to eq 'Title'
+  end
+  it 'It should return book author' do
+    expect(@book.author).to eq 'Author'
+  end
```
Comment (Deepakdanger) [inline spec/book_spec.rb]: - [x] Same goes for this. I would recommend adding test for all the public method you have for the respective class. 

Hint: 
- Add test for `add_rental` method.

File: spec/classroom_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    @classroom = Classroom.new 'label'
+  end
+
+  it 'classroom is an instance of Classroom class' do
+    expect(@classroom).to be_an_instance_of Classroom
+  end
+
+  it 'Should return classroom label' do
+    expect(@classroom.label).to eq 'label'
+  end
```
Comment (Deepakdanger) [inline spec/classroom_spec.rb]: - [x] I would recommend adding test for all the public method you have for the respective class. 

Hint: 
- Add test for `add_student` method.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @TimmyChan99 & @AkshithaReddy1899 , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight :100: 
:+1: PR is descriptive.

## Required Changes ♻️
_Check the comments under the review._

- [x] I would recommend adding the command to test the application. Hint: `rspec`.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @TimmyChan99 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**